src/libtorch.cc

```diff
@@ -1070,56 +1126,37 @@ ModelInstanceState::ProcessRequests(
             &cuda_copy));
   }
 
-  // Request to retrieve all model outputs. 'output_names' and
-  // 'output_tensors' are parallel vectors and so must be kept in
-  // sync.
-  std::vector<const char*> output_names;
-  std::vector<torch::jit::IValue> output_tensors;
-  if (!all_response_failed) {
-    triton::common::TritonJson::Value ios;
-    TRITONSERVER_Error* err =
-        model_state_->ModelConfig().MemberAsArray("output", &ios);
-    if (err == nullptr) {
-      for (size_t i = 0; i < ios.ArraySize(); i++) {
-        triton::common::TritonJson::Value io;
-        err = ios.IndexAsObject(i, &io);
-        if (err != nullptr) {
-          break;
-        }
-
-        // Use names from ModelConfig by reference since the model
-        // config will persist longer than this inference execution.
-        const char* io_name;
-        size_t io_name_len;
-        err = io.MemberAsString("name", &io_name, &io_name_len);
-        if (err != nullptr) {
-          break;
-        }
-
-        output_names.emplace_back(io_name);
-      }
-    }
-
-    if (err != nullptr) {
-      RESPOND_ALL_AND_SET_TRUE_IF_ERROR(
-          responses, request_count, all_response_failed, err);
-      output_names.clear();
-    }
-  }
-
-// Wait for any in-flight input tensor copies to complete.
+  // If the instance kind is not GPU, we need to synchronize the CUDA stream
+  if (Kind() != TRITONSERVER_INSTANCEGROUPKIND_GPU) {
 #ifdef TRITON_ENABLE_GPU
-  if (cuda_copy) {
-    cudaStreamSynchronize(CudaStream());
-  }
+    if (cuda_copy) {
+      cudaStreamSynchronize(stream_);
+      cuda_copy = false;
+    }
 #endif
+  }
 
+  std::vector<torch::jit::IValue> output_tensors;
   uint64_t compute_start_ns = 0;
-  SET_TIMESTAMP(compute_start_ns);
+
+  RESPOND_ALL_AND_SET_TRUE_IF_ERROR(
+      responses, request_count, all_response_failed,
+      RecordBackendTimestamp(
+          &compute_start_ns,
+          reinterpret_cast<cudaEvent_t*>(&compute_infer_event_)));
 
   // Run...
   if (!all_response_failed) {
-    Execute(&responses, request_count, &input_tensors, &output_tensors);
+    if (Kind() == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
+#ifdef TRITON_ENABLE_GPU
+      at::cuda::CUDAStream stream =
+          at::cuda::getStreamFromExternal(stream_, DeviceId());
```

Comment:
Can we keep this as member variable so we don't construct Torch stream wrapper inflight.